p2pd/main.go

```diff
@@ -208,5 +211,10 @@ func main() {
 		}
 	}
 
+	if *metricsAddr != "" {
+		http.Handle("/metrics", prometheus.Handler())
+		go log.Println(http.ListenAndServe(*metricsAddr, nil))
```

Comment:
what's this Println doing?

Reply:
if the http server terminates with an error, it will be logged

Reply:
That's fine, but I am a little confused by the semantics of this. It seems that only the `Println` will be executed in the goroutine, the `listenAndServe` will be synchronous -- which isn't what you want.

Reply:
ahhhh i *think* the entire block is run asynchronously but i need to confirm. let me test. thanks for pointing it out!

Reply:
yep you nailed it. fixed.